packages/forms/src/model.ts

```diff
@@ -430,27 +428,119 @@ export abstract class AbstractControl {
    * When you add or remove a validator at run time, you must call
    * `updateValueAndValidity()` for the new validation to take effect.
    *
+   * If you want to add a new validator without affecting existing ones, consider
+   * using `addValidators` method instead.
+   */
+  setValidators(validators: ValidatorFn|ValidatorFn[]|null): void {
+    this._rawValidators = validators;
+    this._composedValidatorFn = coerceToValidator(validators);
+  }
+
+  /**
+   * Sets the asynchronous validators that are active on this control. Calling this
+   * overwrites any existing asynchronous validators.
+   *
+   * When you add or remove a validator at run time, you must call
+   * `updateValueAndValidity()` for the new validation to take effect.
+   *
+   * If you want to add a new validator without affecting existing ones, consider
+   * using `addAsyncValidators` method instead.
+   */
+  setAsyncValidators(validators: AsyncValidatorFn|AsyncValidatorFn[]|null): void {
+    this._rawAsyncValidators = validators;
+    this._composedAsyncValidatorFn = coerceToAsyncValidator(validators);
+  }
+
+  /**
+   * Add a synchronous validator or validators to this control, without affecting other validators.
+   *
+   * When you add or remove a validator at run time, you must call
+   * `updateValueAndValidity()` for the new validation to take effect.
+   *
+   * Adding a validator that already exists will have no effect.
+   *
+   * @param validators The new validator function or functions to add to this control.
    */
-  setValidators(newValidator: ValidatorFn|ValidatorFn[]|null): void {
-    this._rawValidators = newValidator;
-    this._composedValidatorFn = coerceToValidator(newValidator);
+  addValidators(validators: ValidatorFn|ValidatorFn[]): void {
+    const newValidators = makeValidatorsArray(validators).filter(v => !this.hasValidator(v));
+    const oldValidators = makeValidatorsArray(this._rawValidators);
+    this.setValidators(oldValidators.concat(newValidators));
   }
```

Comment:
I think we can compress it a bit more to avoid duplication between sync and async validators logic:
```suggestion

  // outside of the AbstractControl class...
  function addValidators<T>(newValidators: T|T[], currentValidators: T|T[]|null): T[] {
    const _new = makeValidatorsArray(newValidators).filter(v => !hasValidator(currentValidators, v));
    const _current = makeValidatorsArray(currentValidators);
    return _current.concat(_new);
  }

  // inside the AbstractControl class...
  addValidators(validators: ValidatorFn|ValidatorFn[]): void {
    const newValidators = addValidators<ValidatorFn>(validators, this._rawValidators);
    this.setValidators(newValidators);
  }
  
  addAsyncValidators(validators: AsyncValidatorFn|AsyncValidatorFn[]): void {
    const newValidators = addValidators<AsyncValidatorFn>(validators, this._rawAsyncValidators);
    this.setValidators(newValidators);
  }
```

and similarly we can update remove validators function to extract common duplicated logic.